pkg/webhook/server/webhook_cert_init.go

```diff
@@ -0,0 +1,228 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package server
+
+import (
+	"context"
+	"errors"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+
+	"github.com/che-incubator/che-workspace-operator/internal/controller"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	secureServiceName = "che-workspace-controller-secure-service"
+)
+
+// InitWebhookServer setups TLS for the webhook server
+func InitWebhookServer(ctx context.Context) error {
+	crclient, err := controller.CreateClient()
+	if err != nil {
+		return err
+	}
+
+	ns, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		return err
+	}
+
+	err = syncService(ctx, crclient, ns)
+	if err != nil {
+		return err
+	}
+
+	err = syncConfigMap(ctx, crclient, ns)
+	if err != nil {
+		return err
+	}
+
+	err = updateDeployment(ctx, crclient, ns)
+	if err != nil {
+		return err
+	}
+
+	return errors.New("TLS is setup. Controller needs to restart to apply changes")
+}
+
+func syncService(ctx context.Context, crclient client.Client, namespace string) error {
+	secureService := getSecureServiceSpec(namespace)
+	if err := crclient.Create(ctx, secureService); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+		existingCfg := &v1.Service{}
+		err := crclient.Get(ctx, types.NamespacedName{
+			Name:      secureService.Name,
+			Namespace: secureService.Namespace,
+		}, existingCfg)
+
+		err = crclient.Update(ctx, secureService)
+		if err != nil {
+			return err
+		}
+		log.Info("Updated secure service")
+	} else {
+		log.Info("Created secure service")
+	}
+	return nil
+}
+
+func getSecureServiceSpec(namespace string) *v1.Service {
+	label := map[string]string{"app": "che-workspace-controller"}
```

Comment:
It would be much safer if we make sure that deployment really contains such label and probably provision if missing.